### DISCUSSION TITLE
Pausing during debuging doesn't show next statement

### DISCUSSION THREAD

**Discussion by jimmyxXx** [dnSpyEx/dnSpy#203] [Q&A]
I started an 32-bit .net .exe using dnSpy, and it runs fine. If a breakpoint that I set is hit, execution correctly pauses. However, if I hit the pause button in dnSpy, execution pauses but dnSpy doesn't show the code where execution is paused. If I click the "next statement" button, nothing happens. Is this a known limitation, or am I doing something wrong?

**Comment by ElektroKill** [comment]
> I started an 32-bit .net .exe using dnSpy, and it runs fine. If a breakpoint that I set is hit, execution correctly pauses. However, if I hit the pause button in dnSpy, execution pauses but dnSpy doesn't show the code where execution is paused. If I click the "next statement" button, nothing happens. Is this a known limitation, or am I doing something wrong?

Without more information like exact reproduction instructions, I can only provide a guess as to what may be happening. If you click the pause button and the default thread is currently inside managed code (a transition to native code has occurred and the native code has not yet returned back to the managed code), dnSpy will just pause but won't be able to navigate to the statement since it resides in native code. You can inspect the current call stack in the call stack window and see if the code is currently in a native transition and look at the Threads window to see if there are any other active threads executing managed code you might be interested in. It is also possible that the default thread is suspended at the time you pressed Pause which might cause this behavior to occur. In that case please check the Threads window and try switching to other threads to see if you can find a thread inside managed code.

Hope this answer helps you with the problem. If you still face troubles, I will need some detailed instructions on how to reproduce this problem on my system (including a target to debug if applicable). If you do not feel comfortable with sharing any part of the instructions to reproduce the issue publically, please sent them to me via email: elektrokilldev@protonmail.com

**Reply by jimmyxXx** [reply]
I'm giving this another try :^) 
When pausing debugging, even when paused inside managed code and I can step through code, I can't seem to see the call stack window nor the threads window; I can see "Search", "Locals", and "C# Interactive". I can also make the "Output" window show up through the View menu. How do I show the Call stack and Threads windows?
Thanks!

**Reply by riQQ** [reply]
Iirc they are in the menu `Debug > Windows` or something similar.

**Reply by jimmyxXx** [reply]
You would think so, but there are no items to display call stack or threads windows under Debug nor View...

**Reply by riQQ** [reply]
They are only available in the menu during debugging iirc. They are not shown in the menu directly after starting dnSpy.
